### PR TITLE
Adding additional support for migrating pem to certstore

### PR DESCRIPTION
**Pull request by johnmccrae**
Signed-off-by: John McCrae <john.mccrae@progress.com>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: lib/chef/client.rb [hunk 1]
```diff
@@ -715,11 +718,57 @@ def self.generate_pfx_package(cert_name, date)
...
+      # Chef client and node objects exist on Chef Server already
+      # Create a new public/private keypair in secure storage
+      # and register the new public cert with Chef Server
       require "time" unless defined?(Time)
       autoload :URI, "uri"
 
-      # KeyMigration.instance.key_migrated = true
+      node = Chef::Config[:node_name]
+      d = Time.now
+      if d.month == 10 || d.month == 11 || d.month == 12
```
Comment (neha-p6) [inline lib/chef/client.rb:730]: @johnmccrae  Should we add some comment as to why these months should be checked and why the specific calculation for `end_date` under if-else block?

**Review by neha-p6 — CHANGES_REQUESTED** [review]

File: lib/chef/client.rb [hunk 2]
```diff
@@ -715,11 +718,57 @@ def self.generate_pfx_package(cert_name, date)
...
+        end_date = Time.new(d.year + 1, d.month - 9, d.day, d.hour, d.min, d.sec).utc.iso8601
```
Comment (marcparadise) [inline lib/chef/client.rb:731]: I think we can also simplify with something like `Time.new(...) + (3600  * 24 * 90)` for 90 days.

File: lib/chef/client.rb (hunk 1 above)
Comment (johnmccrae) [reply lib/chef/client.rb:730]: resolved by redoing the date calcuation as marc suggested. Thanks for the feedback. This was a great idea.

File: lib/chef/client.rb (hunk 2 above)
Comment (johnmccrae) [reply lib/chef/client.rb:731]: corrected.

**Review by PrajaktaPurohit — APPROVED** [review]

**Comment by johnmccrae** [comment]
This thing is super polluted - closing

**Pull request closed without merging**